### PR TITLE
optimize gp flag and fix bug

### PR DESCRIPTION
**Pull request by LeoGuo621**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [x] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -667,7 +668,7 @@ func NewOKExChainApp(
 	enableAnalyzer := sm.DeliverTxsExecMode(viper.GetInt(sm.FlagDeliverTxsExecMode)) == sm.DeliverTxsExecModeSerial
 	trace.EnableAnalyzer(enableAnalyzer)
 
-	if appconfig.GetOecConfig().GetEnableDynamicGp() {
+	if appconfig.GetOecConfig().GetDynamicGpMode() != 2 {
```
Comment (zjg555543) [inline app/app.go]: 2 is a magic number, can use const val?

**Review by zjg555543 — DISMISSED** [review]

File: app/gasprice/gasprice_test.go [hunk 2]
```diff
@@ -75,40 +75,42 @@ func TestOracle_RecommendGP(t *testing.T) {
 				gpo.CurrentBlockGPs.Update(gp, 3500)
 				delta--
 			}
-			gpo.BlockGPQueue.Push(gpo.CurrentBlockGPs)
+			cp := gpo.CurrentBlockGPs.Copy()
+			gpo.BlockGPQueue.Push(cp)
 			testRecommendGP = gpo.RecommendGP()
 			gpo.CurrentBlockGPs.Clear()
 			require.NotNil(t, testRecommendGP)
```
Comment (zjg555543) [inline app/gasprice/gasprice_test.go:82]: can refine the ut expected result, like 		require.Equal(t, value)

File: app/app.go (hunk 1 above)
Comment (LeoGuo621) [reply app/app.go]: done

File: app/gasprice/gasprice_test.go (hunk 2 above)
Comment (LeoGuo621) [reply app/gasprice/gasprice_test.go:82]: I will refine this ut in the near future.

**Review by KamiD — APPROVED** [review]

**Pull request merged**